node/resolvers/Routes/index.ts

```diff
@@ -70,12 +70,103 @@ const Index = {
         route: {
           params: { orderId },
         },
+        storeUserAuthToken,
+        sessionToken,
+        logger,
       },
-      clients: { oms },
+      clients: { vtexId, session, oms, storefrontPermissions },
     } = ctx
 
+    if (!orderId) {
+      throw new UserInputError('Order ID is required')
+    }
+
+    const token: any = storeUserAuthToken
+
+    if (!token) {
+      throw new ForbiddenError('Access denied')
+    }
+
+    const authUser = await vtexId.getAuthenticatedUser(token)
+
+    const sessionData = await session
+      .getSession(sessionToken as string, ['*'])
+      .then((currentSession: any) => {
+        return currentSession.sessionData
+      })
+      .catch((error: any) => {
+        logger.error({
+          error,
+          message: 'orders-getSession-error',
+        })
+
+        return null
+      })
+
+    const profileEmail = sessionData?.namespaces?.profile?.email?.value
+
+    const checkPermissionAgainstOrder = (
+      permissions: string[],
+      orderData: any
+    ) => {
```

Comment:
I strongly believe that could be outside of the Index object once this function is like an if wrapper, what do you think?

Reply:
I'll have to refactor the function to take additional arguments, but I agree it's a good idea

Reply:
Done